common/shell-local/run.go

```diff
@@ -58,31 +59,50 @@ func Run(ui packer.Ui, config *Config) (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		ui.Say(fmt.Sprintf("Running local shell script: %s", script))
 
-		comm := &Communicator{
-			ExecuteCommand: interpolatedCmds,
+		// Check if command can execute against runtime.
+		runCommand := false
+		if len(config.OnlyOn) > 0 {
+			for _, os := range config.OnlyOn {
+				if os == runtime.GOOS {
+					runCommand = true
```

Comment:
Once you find a match, you should probably break out of this loop.

Reply:
Good idea and fixed!